backend/src/queues/secret-scanning/githubScanFullRepository.ts

```diff
@@ -22,26 +24,77 @@ type TScanPushEventQueueDetails = {
 
 githubFullRepositorySecretScan.process(async (job: Job, done: Queue.DoneCallback) => {
   const { organizationId, repository, installationId }: TScanPushEventQueueDetails = job.data
+  const [owner, repo] = repository.fullName.split("/");
+
   try {
     const octokit = new ProbotOctokit({
         auth: {
         appId: await getSecretScanningGitAppId(),
         privateKey: await getSecretScanningPrivateKey(),
         installationId: installationId
-        },
+      },
     });
+
+    // Scan the .infisicalignore file (if it exists) & extract the fingerprints
+    const infisicalIgnoreFileContents = await checkIfInfisicalIgnoreFile(octokit, owner, repo);
+    const batchUpdateOperations: any[] = [];
     const findings : SecretMatch[] = await scanFullRepoContentAndGetFindings(octokit, installationId, repository.fullName)
+
     for (const finding of findings) {
-      await GitRisks.findOneAndUpdate({ fingerprint: finding.Fingerprint}, 
-        {
-        ...convertKeysToLowercase(finding),
-        installationId: installationId,
-        organization: organizationId,
-        repositoryFullName: repository.fullName,
-        repositoryId: repository.id
-      }, {
-        upsert: true
-      }).lean()
+
+      // Create a SHA3-512 hash of the secret & immediately clear the plaintext secret from memory
+      // Used in push events to collate all SHA-3 hashes of the secret in the same repo's file (prevents case where the secret moves)
+      // Also useful to append to the fingerprint (eventually) & for user audit secret scanning log table .csv download
+      let secret = finding.Secret;
+      const sha512Hash = createHash("sha3-512");
+      sha512Hash.update(secret);
+      const hashResult = sha512Hash.digest("hex");
+      secret = "";
```

Comment:
I don't think this is necessary because garbage collection should take care of it

Reply:
You're right - only added this to ensure the secret is overridden as soon as it's not needed.